relays/ethereum/src/ethereum_exchange.rs

```diff
@@ -221,11 +227,18 @@ impl TargetClient<EthereumToSubstrateExchange> for SubstrateTransactionsTarget {
 
 	async fn filter_transaction_proof(&self, proof: &EthereumTransactionInclusionProof) -> Result<bool, Self::Error> {
 		// let's try to parse transaction locally
-		let parse_result = bridge_node_runtime::exchange::EthTransaction::parse(&proof.proof[proof.index as usize]);
+		let (raw_tx, raw_tx_receipt) = &proof.proof[proof.index as usize];
```

Comment:
Do we verify somewhere that `proof.index < proof.proof.len()`?

Reply:
We verify it where it comes from untrusted source - i.e. when verifying proof in runtime module. Here it is relay talking to relay, so it is kind of trusted.